features/fields.feature

```diff
@@ -82,97 +74,69 @@ Functionaliteit: Aanpasbare representatie met de fields parameter
-    En wordt in _links geen enkel ander attribuut dan self en kinderen teruggegeven
-    En is elke link kinderen een geldige uri
-    En wordt er geen gerelateerde sub-resource teruggegeven in _embedded
-
-    Als een ingeschreven persoon wordt geraadpleegd met fields=burgerservicenummer,naam,partners,ouders
-    Dan wordt attribuut _links.partners teruggegeven
-    En wordt attribuut _links.ouders teruggegeven
-    En wordt attribuut burgerservicenummer teruggegeven
-    En wordt attribuut naam.voornamen teruggegeven
-    En wordt attribuut naam.geslachtsnaam teruggegeven
-    En wordt attribuut naam.voorvoegsel teruggegeven
-    En wordt attribuut _links.self teruggegeven
-    En wordt in _links geen enkel ander attribuut dan self, partners en ouders teruggegeven
-    En is elke link partners een geldige uri
-    En is elke link ouders een geldige uri
     En wordt er geen gerelateerde sub-resource teruggegeven in _embedded
 
   Scenario: Gebruik van de fields parameter heeft geen invloed op embedded sub-resources
-    Als een ingeschreven persoon wordt geraadpleegd met fields=partners&expand=kinderen
+    Als een ingeschreven persoon wordt geraadpleegd met fields=geboorte.land&expand=kinderen
+    Dan wordt attribuut geboorte.land teruggegeven
+    En wordt attribuut _links.self teruggegeven
+    En wordt in _links geen enkel ander attribuut dan self teruggegeven
+    En wordt geen enkel ander attribuut dan geboorte, _links en _embedded teruggegeven
+    En wordt attribuut _embedded.kinderen teruggegeven
+    En bevat elk voorkomen van _embedded.kinderen attribuut burgerservicenummer met een waarde
+    En bevat elk voorkomen van_embedded.kinderen attribuut naam met een waarde
+    En bevat elk voorkomen van_embedded.kinderen attribuut geboorte.datum met een waarde
+    En bevat elk voorkomen van_embedded.kinderen attribuut geboorte.plaats met een waarde
+    En bevat elk voorkomen van_embedded.kinderen attribuut geboorte.land met een waarde
+
+  Scenario: Gebruik van de expand parameter heeft geen invloed op de inhoud van de resource
+    Als een ingeschreven persoon wordt geraadpleegd met fields=_links.partners&expand=kinderen
     Dan wordt attribuut _links.partners teruggegeven
     En wordt attribuut _links.self teruggegeven
+    En bevat _links niet attribuut kinderen
     En wordt in _links geen enkel ander attribuut dan self en partners teruggegeven
     En wordt attribuut _embedded.kinderen teruggegeven
     En wordt in _embedded geen enkel ander attribuut dan kinderen teruggegeven
     En wordt geen enkel ander attribuut dan _links en _embedded teruggegeven
 
-    Als een ingeschreven persoon wordt geraadpleegd met fields=naam&expand=kinderen
-    Dan wordt attribuut naam teruggegeven
+  Scenario: Vragen van specifieke velden met de expand parameter heeft geen invloed op de inhoud van de resource, alleen op de inhoud van de embedded subresource
+    Als een ingeschreven persoon wordt geraadpleegd met expand=kinderen.naam.voornamen
```

Comment:
Volgens mij mis ik hier fields=burgerservicenummer,naam,geboorte. Anders zou ik in de Dan stappen ook de andere properties verwachten zoals geheimhoudingPersoonsgegevens,leeftijd etc

Reply:
ik wilde hier benadrukken dat velden die door de expand niet in de subresource kinderen worden geleverd, wel in de ingeschrevenpersonen resource mee worden geleverd. Daarom heb ik specifiek velden genoemd die zowel in de ingeschrevenpersonen als in kinderen voor kunnen komen. Is dat verwarrend?

Reply:
ik heb de fields nu ook toegevoegd